src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1226,8 +1242,7 @@ async ValueTask<NpgsqlDataReader> ExecuteReaderAsync(CommandBehavior behavior, b
                         // to prevents a dependency on the thread pool (which would also trigger deadlocks).
                         // The WriteBuffer notifies this command when the first buffer flush occurs, so that the
                         // send functions can switch to the special async mode when needed.
-                        //TODO: pass the token only if the cancellation was not successful
-                        var sendTask = NonMultiplexingWriteWrapper(connector, async, CancellationToken.None);
+                        var sendTask = NonMultiplexingWriteWrapper(connector, async, cts?.Token ?? cancellationToken);
```

Comment:
If I'm reading the code correctly, if the user-provided ct is cancelable, then cts will be non-null. If that's the case, we should pass CancellationToken.None for clarity. We can also extract this expression to a local variable above to avoid repeating it below.